sxw_resource.c

```diff
@@ -300,101 +263,101 @@ static void _transp_contribution_by_group(RealF use_by_group[]) {
-        RealF ssqr = (transp_ratio - transp_window.ratio_average) * (transp_ratio - transp_window.ratio_average);
+        RealF ssqr = (transp_ratio - transp_window->ratio_average) * (transp_ratio - transp_window->ratio_average);
         // add the new sum of squares, subtract the old.
-        transp_window.sum_of_sqrs += ssqr - transp_window.SoS_array[transp_window.oldest_index];
+        transp_window->sum_of_sqrs += ssqr - transp_window->SoS_array[transp_window->oldest_index];
         // update the sum of squares window.
-        transp_window.SoS_array[transp_window.oldest_index] =  ssqr;
+        transp_window->SoS_array[transp_window->oldest_index] =  ssqr;
         //calculate the standard deviation
-        transp_ratio_sd = final_running_sd(transp_window.size, transp_window.sum_of_sqrs);
+        transp_ratio_sd = final_running_sd(transp_window->size, transp_window->sum_of_sqrs);
     }
 
-    //printf("Year %d: ratio = %f, mean = %f, sos = %f sd = %f\n",Globals.currYear,
-              // transp_ratio,transp_window.ratio_average, transp_window.sum_of_sqrs, transp_ratio_sd);
+    //printf("Year %d: ratio = %f, mean = %f, sos = %f sd = %f\n",Globals->currYear,
+              // transp_ratio,transp_window->ratio_average, transp_window->sum_of_sqrs, transp_ratio_sd);
 
     // If this year's transpiration is notably low (1 sd below the mean), add additional transpired water
-    if (transp_ratio < (transp_window.ratio_average - 1 * transp_ratio_sd)) {
-        //printf("Year %d below 1 sd: ratio = %f, average = %f, sd = %f\n", Globals.currYear,transp_ratio,
-                                                          //transp_window.ratio_average, transp_ratio_sd);
-            RealF min = transp_window.ratio_average - transp_ratio_sd;
-            RealF max = transp_window.ratio_average + transp_ratio_sd;
+    if (transp_ratio < (transp_window->ratio_average - 1 * transp_ratio_sd)) {
+        //printf("Year %d below 1 sd: ratio = %f, average = %f, sd = %f\n", Globals->currYear,transp_ratio,
+                                                          //transp_window->ratio_average, transp_ratio_sd);
+            RealF min = transp_window->ratio_average - transp_ratio_sd;
+            RealF max = transp_window->ratio_average + transp_ratio_sd;
 
             // This transpiration will be added
-            added_transp = (1 - transp_ratio / RandUniFloatRange(min, max, &resource_rng)) * transp_window.average;
-            if(added_transp < 0){
+            transp_window->added_transp = (1 - transp_ratio / RandUniFloatRange(min, max, &resource_rng)) * transp_window->average;
+            if(transp_window->added_transp < 0){
                 LogError(logfp, LOGNOTE, "sxw_resource: Added transpiration less than 0.\n");
             }
-            //printf("Year %d:\tTranspiration to add: %f\n",Globals.currYear,add_transp);
+            //printf("Year %d:\tTranspiration to add: %f\n",Globals->currYear,add_transp);
```

Comment:
This line can probably safely be removed.

Reply:
@fdpierson I'm going to leave this print statement and remove the others, just so we have one for quick debugging.